dbt/include/bigquery/macros/etc.sql

```diff
@@ -6,6 +6,13 @@
   {% do adapter.grant_access_to(entity, entity_type, role, grant_target_dict) %}
 {% endmacro %}
 
+{#
+  This macro returns the partition metadata for provided table.
+  The expected input is a table object (ie through a `source` or `ref`).
+  The output contains the result from partitions information for your input table.
+  The details of the retrieved columns can be found on https://cloud.google.com/bigquery/docs/managing-partitioned-tables
+  It will leverage the INFORMATION_SCHEMA.PARTITIONS table.
+#}
 {%- macro get_partitions_metadata(table) -%}
```

Comment:
I was hoping to add an option to keep the previous approach but I can't since the prototype of `adapter.get_partitions_metadata` is set in dbt-core and I don't think we want to leak the BQ specific option.
I think that, overall, it's acceptable.